src/api/drone.rs

```diff
@@ -1,5 +1,20 @@
-use crate::types::packet::Packet;
+use crate::types::channel::Channel;
+use crate::types::source_routing_header::NodeId;
+use std::collections::HashMap;
+use std::thread;
 
-pub trait DroneAble {
-    fn forward_packet(&self, packet: Packet) -> bool;
+pub struct Drone {
+    id: NodeId,
+    thread: thread::JoinHandle<()>,
```

Comment:
Not sure if thread should be inside of it or not. I'm not perfectly up to speed with thread in rust but it seem to me to have the drone not think about being a thread.

Reply:
The idea was that while the thread needs to have at its disposal the channels, you also need to store them elsewhere for where you want to add a new drone with the Simulation Controller.
Having a struct Drone like this allows to keep track of the list of channels of each drone in an ordered way, and also save a reference to the drone (that might be needed for the crash, but I'm not sure about how do to that still).